File: examples/latest/navigate/android/Java/HERESDKUnits/here-sdk-units-speedlimit/src/main/java/com/here/sdk/units/speedlimit/SpeedLimitView.java
/*
 * Copyright (C) 2019-2025 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */
package com.here.sdk.units.speedlimit;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.TextView;

// The view hosting this HERE SDK Unit.
// The view consists only of a label with speed limit sign.
// The unit is created after inflation and provides additional logic.
public class SpeedLimitView extends LinearLayout {

    public SpeedLimitUnit speedLimitUnit;

    public SpeedLimitView(Context context) {
        super(context);
        init(context);
    }

    public SpeedLimitView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public SpeedLimitView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init (context);
    }

    private void init(Context context) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View layout = inflater.inflate(R.layout.speed_limit_view, this, true);

        // Get references to UI elements
        TextView labelTextView = layout.findViewById(R.id.speed_limit_label);
        TextView speedValueTextView = layout.findViewById(R.id.speed_limit_value);
        FrameLayout outerCircle = layout.findViewById(R.id.speed_limit_outer_circle);
        speedLimitUnit = new SpeedLimitUnit(labelTextView, speedValueTextView, outerCircle);
    }
}

File: examples/latest/navigate/android/Kotlin/HERESDKUnitsKotlin/here-sdk-units-speedlimit/src/main/java/com/here/sdk/units/speedlimit/SpeedLimitView.kt
/*
 * Copyright (C) 2019-2025 HERE Europe B.V.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * License-Filename: LICENSE
 */

package com.here.sdk.units.speedlimit

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// The view hosting this HERE SDK Unit.
// The view consists only of a label with speed limit sign.
// The unit is injected and provides additional logic.
@Composable
fun SpeedLimitView(
    unit: SpeedLimitUnit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Label.
        Text(
            text = unit.uiState.label,
            fontSize = 12.sp,
            fontWeight = FontWeight.Bold,
            color = Color.Black,
            textAlign = TextAlign.Center,
            modifier = Modifier
                .height(17.dp)
        )

        Spacer(modifier = Modifier.height(4.dp))

        // Outer red circle.
        Box(
            modifier = Modifier
                .size(50.dp)
                .clip(CircleShape)
                .background(Color.Red),
            contentAlignment = Alignment.Center
        ) {
            // Inner white circle.
            Box(
                modifier = Modifier
                    .size(37.dp)
                    .clip(CircleShape)
                    .background(Color.White),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = unit.uiState.speedText,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
            }
        }
    }
}
